tests/util/test_sendPREMISEvent.py

```diff
@@ -51,11 +83,34 @@ def test_raises_exception_without_201_status(monkeypatch):
     """
     response = Mock(code=200)
 
-    expected = (response, 'Fake content')
+    expected = (response, b'Fake content')
     doWebRequest = Mock(return_value=expected)
     monkeypatch.setattr('codalib.util.doWebRequest', doWebRequest)
 
     with pytest.raises(Exception):
         util.sendPREMISEvent('http://example.com', None, None, None, None)
 
     assert doWebRequest.call_count == 1
+
+
+def test_raises_exception_without_201_status_with_debug(monkeypatch):
+    """
+    Verify that response content is written to file when sendPREMISEvent
+    will raise an exception when the returned response does not have a
+    status code 201 with debug=True.
+    """
+    response = Mock(code=200)
+
+    expected = (response, b'Fake content')
+    doWebRequest = Mock(return_value=expected)
+    monkeypatch.setattr('codalib.util.doWebRequest', doWebRequest)
+
+    m = mock_open()
+    monkeypatch.setattr('builtins.open', m)
+
+    with pytest.raises(Exception):
+        util.sendPREMISEvent('http://example.com', None, None, None, None, debug=True)
+
+    calls = [call().write(b'Fake content'), call().close()]
+    m.assert_has_calls(calls, any_order=True)
```

Comment:
I don't think we need `any_order=True`, since we would want the `write` to come before the `close`.

Reply:
ok. I'll remove it.

Reply:
Another thing to note, I think it is best to generally be more specific with patching. Monkeypatching `builtins.open` could have unintended consequences patching at that broad of a level. I think you could patch `open` specifically in the module under test and it would be safer:
```
        with patch('codalib.util.open', m):
            util.sendPREMISEvent('http://example.com', None, None, None, None, debug=True)
```